javaagent-core/src/main/java/org/hypertrace/agent/core/config/HypertraceConfig.java

```diff
@@ -64,10 +71,18 @@ private HypertraceConfig() {}
 
   public static AgentConfig get() {
     if (agentConfig == null) {
+      String dynamicConfigServiceUrl =
+          EnvironmentConfig.getProperty(EnvironmentConfig.DYNAMIC_CONFIG_SERVICE_URL);
+      DynamicConfigFetcher dynamicConfigFetcher = null;
+      if (dynamicConfigServiceUrl != null) {
+        dynamicConfigFetcher = new DynamicConfigFetcher(dynamicConfigServiceUrl);
+        Executors.newScheduledThreadPool(1)
```

Comment:
Is there an executor service in the upstream that we could use? The upstream javaagent-tooling might have smth that we could reuse. We also have to set the thread to the daemon otherwise it will be blocking shutdown

Reply:
Will check this

Reply:
Upstream does not have any executor service that I could find. They have also used Executors directly https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SupportabilityMetrics.java#L81